packages/material-ui-lab/src/DayPicker/DayPicker.test.tsx

```diff
@@ -55,7 +55,9 @@ describe('<DayPicker />', () => {
   });
 
   it('switches between views uncontrolled', () => {
-    render(<DayPicker date={adapterToUse.date('2019-01-01T00:00:00.000')} onChange={() => {}} />);
+    render(<DayPicker date={adapterToUse.date('2019-01-01T00:00:00.000')} onChange={() => {}} />, {
+      strict: false, // findDOMNode issue
+    });
```

Comment:
I should really just remove this option. `createRender({ strict })` is meant to signal if the component as a whole is StrictMode compatible. Disabling them per test just hides issues instead of fixing them.

Reply:
> Disabling them per test just hides issues instead of fixing them.

I would advocate that this option allows enforcing more constraints on the component. In the material-ui-x repository. The grid is strict mode compatible with the exception of one feature: the apiRef #849. We run most of the tests in strict mode with the exception of this very feature. I think that it's important to be able to run as many tests as possible in strict mode, the more constraints, the less likely we can introduce regression, inside a given component. This is not specific to the data grid, we have the same with the date picker, I believe.

Reply:
Why is it hiding something?

Reply:
A component as a whole can only be StrictMode compatible. Everything else is not a useful signal for users

> We call createRender twice, create two renders, one with strict and one without. => more visibility

Same issue. 

There's no reason to do this right now. The old approach was working perfectly fine. You have a choice of keeping this approach. Following my previous work that I've been mostly responsible for. Or changing it for made up reasons.

Reply:
@eps1lon Ok, would the opposite work with you? strict: false at the root and, strict: true, in individual test cases.

If not, should I revert 

https://github.com/eps1lon/material-ui/blob/921b3c6e57bf3258a174da4d47229124765c8727/packages/material-ui/test/integration/Select.test.js#L134

too?